### ISSUE TITLE
Default value for optional feature of "Domain.Attribute" and "Domain.Feature" should be `true`

### ISSUE THREAD

**Issue by gcoutable** [eclipse-sirius/sirius-components#654]
If we look in "domain.ecore" the default value of the eclass _Feature_ feature _Optional_ is `false`. That implies both _Feature_ subclasses, _Relation_ and _Attribute_, have the same default value for their feature _Optional_.
It has been decided the default value for the feature _Optional_ of _Relation_ has to be `true`, which is something Ecore is not able to do with ease. To do so, _DomainFactoryImpl_ has been modified.

It has created an inconsistency between both _Relation_ and _Attribute_ _Optional_ feature, which is not handled by emfjson, use to serialize and deserialize our emf model in/from json.

To summarize, we are not able to make a _Relation_ optional.

### Steps to reproduce

1. In a project, creates a new Domain with an Entity and a Relation
2. Unchecks the optional checkbox
3. In another project, creates an instance of the domain with an entity instance
4. Goes back back to the domain project
5. Displays Relation details

### Expected behavior

The Relation is still optional (the "optional" checkbox is unchecked)

### Actual behavior

The Relation is not optional anymore (the "optional" checkbox is checked)
